File: 2024/src/01/day01a.m
% Open the file and read it line by line
fileID = fopen('input/day01.txt', 'r');

lefties = [];
righties = [];

% Read input
while ~feof(fileID)
    line = fgetl(fileID);
    
    parts = strsplit(strtrim(line), ' ');
    lefties = [lefties, str2double(parts{1})];
    righties = [righties, str2double(parts{2})];
end

% Process
lefties = sort(lefties);
righties = sort(righties);
result = sum(abs(lefties-righties))
fclose(fileID);

File: 2024/src/01/day01a.py
lefties = []
righties = []

# Read input
with open("input/day01.txt", "r") as file:
    for line in file:
        parts = line.strip().split()
        l, r = map(int, parts)
        lefties.append(l)
        righties.append(r)

# Process
lefties.sort()
righties.sort()
result = sum(abs(l - r) for l, r in zip(lefties, righties))
print(result)
